### PR TITLE
Remove restrictions on StrippedState

### PR DESCRIPTION
**Pull request by illicitonion**
Note also that these are purely informational and not things on which to
rely.

SPEC-357

### CODE REVIEW

File: event-schemas/schema/m.room.member [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
...B to join a room, and trying to force this state change directly will fail. \n\nThe following membership states are specified:\n\n- ``invite`` - The user has been invited to join a room, but has not yet joined it. They may not participate in the room until they join.\n\n- ``join`` - The user has joined the room (possibly after accepting an invite), and may participate in it.\n\n- ``leave`` - The user was once joined to the room, but has since left (possibly by choice, or possibly by being kicked).\n\n- ``ban`` - The user has been banned from the room, and is no longer allowed to join it until they are un-banned from the room (by having their membership state set to a value other than ``ban``).\n\n- ``knock`` - This is a reserved word, which currently has no meaning.\n\nThe ``third_party_invite`` property will be set if this invite is an ``invite`` event and is the successor of an ``m.room.third_party_invite`` event, and absent otherwise.\n\nThis event may also include an ``invite_room_state`` key **outside the** ``content`` **key**. If present, this contains an array of ``StrippedState`` Events. These events provide information on a a subset of state events such as the room name.",
```
Comment (ara4n) [inline event-schemas/schema/m.room.member]: s/a a/a/

**Comment by ara4n** [comment]
I suspect we should still recommend the event types that should be included in invites for a good UX by clients when rendering the invite, even if we're removing the restriction.

File: event-schemas/schema/m.room.member (hunk 1 above)
Comment (illicitonion) [inline event-schemas/schema/m.room.member]: Done

**Comment by illicitonion** [comment]
Done

**Comment by ara4n** [comment]
lgtm, other than possibly wondering whether the should on line 71 should be a SHOULD.

**Pull request merged**